heartbeat/scheduler/scheduler.go

```diff
@@ -21,22 +21,29 @@ import (
 	"errors"
 	"runtime/debug"
 	"sync"
+	"sync/atomic"
 	"time"
 
 	"github.com/elastic/beats/libbeat/logp"
 )
 
+const (
+	statePreRunning = int32(iota + 1)
+	stateRunning
+	stateDone
+)
+
 type Scheduler struct {
-	limit   uint
-	running bool
+	limit uint
+	state *int32
 
 	location *time.Location
 
 	jobs   []*job
 	active uint // number of active entries
 
-	add, rm  chan *job
-	finished chan taskOverSignal
+	addCh, rm chan *job
+	finished  chan taskOverSignal
```

Comment:
for naming consistency let's also add `Ch` to `rm` and `finished` ?

Reply:
++